modules/table/src/main/java/org/apache/ignite/internal/table/distributed/index/MetaIndexStatusEnum.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed.index;
+
+import org.apache.ignite.internal.catalog.descriptors.CatalogIndexStatus;
+
+/** Enumeration the index status stored in the {@link IndexMeta}. */
+public enum MetaIndexStatusEnum {
+    /**
+     * Index has been created in the catalog.
+     *
+     * @see CatalogIndexStatus#REGISTERED
+     */
+    REGISTERED,
+
+    /**
+     * Index was in the process of being built.
+     *
+     * @see CatalogIndexStatus#BUILDING
+     */
+    BUILDING,
+
+    /**
+     * Index has been successfully built and is available for reading from it.
+     *
+     * @see CatalogIndexStatus#AVAILABLE
+     */
+    AVAILABLE,
+
+    /**
+     * Available index has started preparing to be removed from the catalog.
+     *
+     * @see CatalogIndexStatus#STOPPING
+     */
+    STOPPING,
+
+    /**
+     * Index has been removed from the catalog, with {@link CatalogIndexStatus#REGISTERED}/{@link CatalogIndexStatus#BUILDING} statuses, but
+     * has not yet been destroyed due to a low watermark.
+     */
+    REMOVED,
```

Comment:
To leave removed from catalog indexes that will not yet be destroyed under LVM. 
For example, in `org.apache.ignite.internal.table.distributed.raft.snapshot.FullStateTransferIndexChooser#chooseForAddWriteCommitted`  and `org.apache.ignite.internal.table.distributed.TableUtils#indexIdsAtRwTxBeginT`s we will take all indexes including removed ones. I think it would be strange when receiving such indexes not to get a meta for them.

Reply:
I mean, if an index is dropped before it became AVAILABLE, we can destroy it right away, we don't need to wait for LWM to come. If we need some special status to mark such a 'already invalid and will be dropped soon' index to make the code more readable (for example), then ok, but we should not write to such an index, too.

Reply:
Yes we can, but it doesn't work yet. When we remove the index from the catalog, we only put it in the queue for destruction until the LVM is updated. We should leave consistent behavior. This will be an optimization and will need to be done more carefully.

See:
org.apache.ignite.internal.index.IndexManager#onIndexRemoved
org.apache.ignite.internal.index.IndexManager#onLwmChanged

Reply:
Ok. Do we have a Jira issue for this optimization? If not, let's file it and add a TODO here.